phoenix-core/src/main/java/org/apache/phoenix/schema/types/PInteger.java

```diff
@@ -272,4 +273,10 @@ public PhoenixArray newArray(PDataType type, Object[] elements) {
       };
     }
   }
+
+  @Override
+  public int getSign(byte[] bytes, int offset, int length, SortOrder sortOrder,
+          Integer maxLength, Integer scale) {
+      return SignUtils.getSignForSignedInteger(bytes, offset, length, sortOrder, Bytes.SIZEOF_INT);
```

Comment:
Same idea here. Create an intermediate class, called PWholeNumber or PWholeNumeric and derive PLong, PInteger, PShort, PByte and all the unsigned variations from this one. Then you can do something like this:

    long l = getCodec().decodeLong(bytes, offset, sortOrder);
    return l == 0 ? 0 : l < 0 ? -1 : 1; // Or maybe a mask of the most significant bit is better?